crates/dbsp/src/operator/sample.rs

```diff
@@ -70,6 +75,57 @@ where
             )
         })
     }
+
+    /// Generates a subset of keys that partition the set of all keys in `self`
+    /// into `num_quantiles + 1` approximately equal-size quantiles.
+    ///
+    /// Internally, this operator uses the
+    /// [`stream_sample_keys`](`Self::stream_sample_keys`) operator to compute a uniforn random
+    /// sample of size `num_quantiles ^ 2` and then picks every `num_quantile`'s element of the
+    /// sample.
+    ///
+    /// Maximal supported `num_quantiles` value is [`MAX_QUANTILES`].  If the operator
+    /// receives a larger `num_quantiles` value, it treats it as
+    /// `MAX_QUANTILES`.
+    ///
+    /// Outputs a Z-set containing `<=num_quantiles` keys.  Each key is output
+    /// with weight `1` regardless of its weight or the number of associated
```

Comment:
is this true even if the stream has 10^6 identical values and a few other values?

Reply:
a correct quantile computation would return many identical values. after all, this is how the distribution looks.

Reply:
This operator assumes set semantics.

Reply:
I think that is not useful

Reply:
Depends what you want to use it for :) It's useful for implementing a slider over the collection.

Reply:
For this you need an api that returns only one value, specified using a numerator and a denominator. Could be much more efficient